cartograph/vector_augmenter.py

```diff
@@ -65,21 +32,60 @@ def create_label_matrix(label_matrix):
+    svd = TruncatedSVD(n_components=10, n_iter=7, random_state=42)
+    label_wide_matrix = create_label_matrix(art_labels)
+    lp_mat_reduced = svd.fit_transform(X=label_wide_matrix.iloc[:, 1:].to_numpy())
+    reduce_vec_labels = ['svd_'+str(i) for i in range(lp_mat_reduced.shape[1])]
+    label_svd = pd.DataFrame({}, columns=['article_id']+reduce_vec_labels)
+    for i in range(len(article_vectors)):
+        label_svd.loc[i, 'article_id'] = article_vectors.loc[i, 'article_id']
+        label_svd.iloc[i, 1:] = lp_mat_reduced[i, :]
+    return label_svd
 
-if __name__ == '__main__':
-    import sys
-    if len(sys.argv) != 3:
-        sys.stderr.write('Usage: %s map_directory augmentation_method' % sys.argv[0])
-        sys.exit(1)
 
-    map_directory, method = sys.argv[1:]
-    main(map_directory, method)
+def get_cluster_matrix(cluster_csv):
+    num_of_countries = len(cluster_csv['country'].unique())
+    cluster_csv = cluster_csv[cluster_csv['country'] != -1]
+    cluster_matrix = np.zeros((cluster_csv.shape[0], num_of_countries + 1))
+    for i in range(len(cluster_csv['article_id'])):
+        cluster_matrix[i][0] = cluster_csv.loc[i, 'article_id']  # assign the first column to be article ids
+        country = cluster_csv.loc[i, 'country']
+        cluster_matrix[i][country+1] = 1
+    country_labels = ['country_' + str(i) for i in range(num_of_countries)]
+    cluster_matrix_df = pd.DataFrame(cluster_matrix, columns=['article_id'] + country_labels)
+    return cluster_matrix_df
 
 
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Augment the original article vectors with label matrix or '
+                                                 'cluster matrix.')
+    parser.add_argument('--experiment', required=True)
+    parser.add_argument('--vectors', required=True)
+    parser.add_argument('--label_vectors', required=True)
+    parser.add_argument('--method', required=True)
+    parser.add_argument('--cluster_vectors', required=True)
+    args = parser.parse_args()
+
+    article_vectors = pd.read_csv(args.vectors)
+    label_csv = pd.read_csv(args.label_vectors)
+    cluster_csv = pd.read_csv(args.cluster_vectors)
+
+    label_vectors = get_label_svd(article_vectors, label_csv)
+    cluster_vectors = get_cluster_matrix(cluster_csv)
+
+    if args.method == 'label':
+        cluster_df = pd.merge(article_vectors, label_vectors, on='article_id')
+    elif args.method == 'cluster':
+        cluster_df = pd.merge(cluster_vectors, article_vectors, on='article_id')
+    elif args.method == 'all':
+        cluster_df_with_cluster = pd.merge(cluster_vectors, article_vectors, on='article_id')
+        cluster_df = pd.merge(cluster_df_with_cluster, label_vectors, on='article_id')
+    else:
+        sys.stderr.write("Unkonwn clustering method: %s\n" + args.clustering)
+        sys.exit(1)
+    cluster_df.to_csv('%s/augmented_vectors.csv' % (args.experiment, ), index=False)
```

Comment:
I think this should take an output file rather than always writing to augmented_vectors.csv.